python/src/interpreter.cc

```diff
@@ -11,6 +11,8 @@ namespace py = pybind11;
 
 namespace {
 
+typedef uint16_t npy_half;
```

Comment:
Can you make this a distinct type so the overloads won't break if we ever add `uint16_t` atomics.

Reply:
You just need to specialize `OpCreator::create` for `npy_half` so it doesn't call `pybind11::dtype::of<npy_half>`.

Reply:
> Can you make this a distinct type so th

indeed, done